lib/spack/spack/cmd/__init__.py

```diff
@@ -174,20 +174,25 @@ def elide_list(line_list, max_num=10):
         return line_list
 
 
-def disambiguate_spec(spec, env, local=False):
+def disambiguate_spec(spec, env, local=False, installed=True):
     """Given a spec, figure out which installed package it refers to.
 
     Arguments:
         spec (spack.spec.Spec): a spec to disambiguate
         env (spack.environment.Environment): a spack environment,
             if one is active, or None if no environment is active
         local (boolean, default False): do not search chained spack instances
+        installed (iterable or boolean, default True): iterable of install
+            status to match. If true, use ['installed']. If false, use
+            ['missing'].
```

Comment:
These details like "true -> ['installed']" are I think too-detailed for this docstring: these are details handled by the query method (it could make sense to redirect the user there to understand install status). So something like

```
installed: install_status or iterable of install statuses, as defined in Database.query
```

would avoid confusion I think